sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -873,8 +873,7 @@ static int _match_to_idx(const gnrc_netif_t *netif,
             continue;
         }
         match = ipv6_addr_match_prefix(&(netif->ipv6.addrs[i]), addr);
-        if (((match > 64U) || !ipv6_addr_is_link_local(&(netif->ipv6.addrs[i]))) &&
-            (match >= best_match)) {
+        if (match >= best_match) {
```

Comment:
If you set this to `>` instead of `>=` the test passes. Reasoning: `best_match` is initialized with `0`, but the addresses don't match, so `ipv6_addr_match_prefix(&(netif->ipv6.addrs[i]), addr);` returns 0, so this check still succeeds and the function returns index 0 for the address array instead of `-1` (no matching address found). Even for non-link-local addresses this might be the reason for some bugs seen in the past.

Reply:
For the correct case its arbitrary if `>=` or `>` is used, as the order of the addresses in the array has no meaning (they might be in the order they were added, but if an address is deleted and than another one added in its spot this isn't the case anymore).

Reply:
yes, done